devolutions-gateway/src/api/monitoring.rs

```diff
@@ -0,0 +1,224 @@
+use crate::DgwState;
+use crate::http::HttpError;
+use axum::{Json, Router, extract, routing};
+use network_monitor;
+use time::OffsetDateTime;
+
+pub fn make_router<S>(state: DgwState) -> Router<S> {
+    let router = Router::new()
+        .route("/config", routing::post(handle_set_monitoring_config))
+        .route("/log/drain", routing::post(handle_drain_log));
+
+    router.with_state(state)
+}
+
+/// Replace the current monitoring configuration with the configuration in the request body.
+///
+/// Changes take effect immediately:
+/// - Starts any monitors newly defined in the payload.
+/// - Stops any currently running monitors that are omitted from the payload.
+///
+/// Note: The configuration is not persisted across process restarts.
+#[cfg_attr(feature = "openapi", utoipa::path(
+    post,
+    operation_id = "SetMonitoringConfig",
+    tag = "NetworkMonitoring",
+    path = "/jet/net/monitor/config",
+    request_body(content = MonitorsConfig, description = "JSON object containing a list of monitors", content_type = "application/json"),
+    responses(
+        (status = 200, description = "New configuration was accepted"),
+        (status = 400, description = "Bad request"),
+        (status = 401, description = "Invalid or missing authorization token"),
+        (status = 403, description = "Insufficient permissions"),
+        (status = 500, description = "Unexpected server error while starting monitors"),
+    ),
+))]
+async fn handle_set_monitoring_config(
+    extract::State(DgwState { monitoring_state, .. }): extract::State<DgwState>,
+    Json(config): Json<MonitorsConfig>,
+) -> Result<Json<SetConfigResponse>, HttpError> {
+    let (processed_config, probe_type_errors) = config.lossy_into();
+
+    network_monitor::set_config(processed_config, monitoring_state)
+        .await
+        .map(|_| Json(SetConfigResponse::new(probe_type_errors)))
+        .map_err(
```

Comment:
style: https://github.com/Devolutions/IronRDP/blob/6fddcaae386cddb32c4bcc3a349bc07308c65a59/STYLE.md#formatting-of-error-messages
```suggestion
                .with_msg("failed to set up network monitoring")
```